internal/transport/transport_test.go

```diff
@@ -2725,7 +2800,7 @@ func (s) TestClientSendsAGoAwayFrame(t *testing.T) {
 		}
 	}()
 
-	ct, err := NewClientTransport(ctx, context.Background(), resolver.Address{Addr: lis.Addr().String()}, ConnectOptions{}, func(GoAwayReason) {})
+	ct, err := NewClientTransport(ctx, context.Background(), resolver.Address{Addr: lis.Addr().String()}, connectOptions, func(GoAwayReason) {})
```

Comment:
why are client things there in createServer? We should probably move only the goroutine part to a new func and call that in the 2 go away tests. Rest everything can remain in the main test

Reply:
Probably better idea to have all the client and server creation in a single place as both of them are common to `Test/ClientSendsAGoAway` and `Test/ClientCloseTimeoutOnHang`. Refactored the func name to `createClientServerConn`.

Reply:
i think only server part is common. For client though, one is using hanging conn and another one is using a regular conn. I would just refactor the server part